### DISCUSSION TITLE
这是一个很奇怪的需求，请大神帮忙看下。

### DISCUSSION THREAD

**Discussion by Atw-Lee** [ant-design/ant-design#41658] [Q&A] [answered]
1、我使用cra初始化了一个项目。

2、我使用了script标签的形式引入了antd的js文件。<script src="https://unpkg.com/antd@4.24.7/dist/antd.min.js"></script>。

3、我如何在不使用script引入react的前提下，获取到应该挂载到window上的antd。目前因为没有引入react文件，产生了如下错误：

Uncaught TypeError: Cannot read properties of undefined (reading 'Component')
    at antd.min.js:sourcemap:23:24611
    at Module.i (antd.min.js:sourcemap:23:24767)
    at a (antd.min.js:sourcemap:24:792267)
    at Object.i (antd.min.js:sourcemap:24:786746)
    at a (antd.min.js:sourcemap:24:792267)
    at Object.i (antd.min.js:sourcemap:24:786705)
    at a (antd.min.js:sourcemap:24:792267)
    at antd.min.js:sourcemap:24:792188
    at antd.min.js:sourcemap:9:323
    at antd.min.js:sourcemap:9:354

DomWrapper.js:23 Uncaught TypeError: Cannot read properties of undefined (reading 'Component')
    at DomWrapper.js:23:3
    at Module.i (DomWrapper.js:23:1)
    at a (bootstrap:19:22)
    at Object.i (index.js:3:18)
    at a (bootstrap:19:22)
    at Object.i (antd.min.js:24:786705)
    at a (bootstrap:19:22)
    at bootstrap:83:10
    at universalModuleDefinition:9:18
    at universalModuleDefinition:10:2

**Comment by yoyo837** [comment]
为什么能使用script引入a, 又不能使用script引入a的依赖项b?

**Comment by yoyo837** [comment]
什么样的场景限制了这个, 只能使用一次script? 那就script到另一个js, 这个js负责引入其他js?

**Comment by RedJue** [comment]
我好奇的是为什么不用npm包，你都用cra初始化项目了

**Comment by Atw-Lee** [comment]
我们基于antd封装了一套业务相关的组件库，npm包也已经维护很久了。不知道什么场景，leader需要使用一个js文件来使用我们的组件库，我使用了father产出了umd模式的js文件，经过实验，使用它需要引入依赖的react及antd，这样是可以正常使用的。但是却不想再次引入react及antd的script文件，因为当前工程已经已经有了antd的npm包，如果多次引入，可能会导致产物加倍的问题存在吧。

**Comment by yoyo837** [comment]
那你就在初始化的地方npm包自己的代码 window.React = xxx 给到window

**Comment by yoyo837** [comment, marked as answer]
谁从window上取什么, 你就负责往 window 上挂什么.

**Reply by Atw-Lee** [reply]
一语惊醒梦中人，十分感谢，我从来没想到手动往window上去挂。思维还是局限了，基础还是不扎实。感谢感谢。
